kube/compose.go

```diff
@@ -20,42 +20,104 @@ package kube
 
 import (
 	"context"
+	"fmt"
+	"strings"
 
 	"github.com/compose-spec/compose-go/types"
 
 	"github.com/docker/compose-cli/api/compose"
+	apicontext "github.com/docker/compose-cli/api/context"
+	"github.com/docker/compose-cli/api/context/store"
 	"github.com/docker/compose-cli/api/errdefs"
+	"github.com/docker/compose-cli/api/progress"
 	"github.com/docker/compose-cli/kube/charts"
+	"github.com/docker/compose-cli/kube/resources"
 )
 
+type composeService struct {
+	sdk *charts.HelmActions
+}
+
 // NewComposeService create a kubernetes implementation of the compose.Service API
 func NewComposeService(ctx context.Context) (compose.Service, error) {
-	chartsAPI, err := charts.NewSDK(ctx)
+	contextStore := store.ContextStore(ctx)
+	currentContext := apicontext.CurrentContext(ctx)
+	var kubeContext store.KubeContext
+
+	if err := contextStore.GetEndpoint(currentContext, &kubeContext); err != nil {
+		return nil, err
+	}
+	config, err := resources.LoadConfig(kubeContext)
+	if err != nil {
+		return nil, err
+	}
+	actions, err := charts.NewHelmActions(config)
 	if err != nil {
 		return nil, err
 	}
 	return &composeService{
-		sdk: chartsAPI,
+		sdk: actions,
 	}, nil
 }
 
-type composeService struct {
-	sdk charts.SDK
-}
-
 // Up executes the equivalent to a `compose up`
 func (s *composeService) Up(ctx context.Context, project *types.Project, options compose.UpOptions) error {
-	return s.sdk.Install(project)
+	w := progress.ContextWriter(ctx)
+
+	eventName := "Convert to charts"
+	w.Event(progress.CreatingEvent(eventName))
+	w.Event(progress.NewEvent(eventName, progress.Working, ""))
+
+	chart, err := charts.GetChartInMemory(project)
+	if err != nil {
+		return err
+	}
+	w.Event(progress.NewEvent(eventName, progress.Done, ""))
+
+	eventName = "Install charts"
```

Comment:
Nit : maybe mention helm here : "Install helm charts". Just making it clear to users this is all helm charts